### PR TITLE
GH Issue 336: additions to builds_and_images_streams file

### PR DESCRIPTION
**Pull request by bfallonf**
@maxamillion @brenton As per #336 . I put in some of the info from the link provided. Is there anything else you think is needed for this issue?

Thanks.

### CODE REVIEW

**Comment by brenton** [comment]
I'll let Adam speak for himself but my suspicion is that he's looking for lower level documentation.  We do have this in the codebase: https://github.com/openshift/origin/blob/master/docs/builds.md.

Documentation at that level is likely unsuitable for the "core concepts" section.

File: architecture/core_concepts/builds_and_image_streams.adoc [hunk 1]
```diff
@@ -11,11 +11,21 @@
 toc::[]
 
 == Builds
-A build is the process of transforming input parameters into a resulting object.
-Most often, the process is used to transform source code into a runnable image.
-A link:../../dev_guide/builds.html#defining-a-buildconfig[BuildConfig] object is
+A link:../../dev_guide/builds.html[build] is the resulting object of
```
Comment (adellape) [inline architecture/core_concepts/builds_and_image_streams.adoc]: This seems like it's losing some of its meaning. We're going from saying a build is a process to now saying it's the resulting object of itself? In the original version, the resulting object was the runnable image.

File: architecture/core_concepts/builds_and_image_streams.adoc [hunk 2]
```diff
@@ -26,13 +36,9 @@ build strategies available:
 
 By default, Docker builds and S2I builds are supported.
 
-The resulting object of a build depends on the builder used to create it. For
```
Comment (adellape) [inline architecture/core_concepts/builds_and_image_streams.adoc]: I don't think we should drop "The resulting object of a" at the start.

File: architecture/core_concepts/builds_and_image_streams.adoc [hunk 3]
```diff
@@ -26,13 +36,9 @@ build strategies available:
...
-Docker and S2I builds, the resulting objects are runnable images. For Custom
-builds, the resulting objects are whatever the builder image author has
-specified.
-
-For a list of build commands, see the
-link:../../dev_guide/builds.html[Developer's Guide].
```
Comment (adellape) [inline architecture/core_concepts/builds_and_image_streams.adoc]: Why remove this?

**Comment by adellape** [comment]
@bfallonf What's your thoughts on this?

**Comment by bfallonf** [comment]
I'd forgotten about this.

@adellape I reverted back some of my edits. The ones you said at least. In my opinion, the docs were a little confused on if it was the process or the resulting object. If it's indeed the process, then i'm thinking this needs to be a little more clear. "Builds" might not be the best word. Maybe "Building"? Or "The Builder"?

@brenton I'm not too sure what you mean by "lower-level documentation". And I believe I placed the info here because it's still conceptual. This topic and the dev_guide/builds topic do link to each other, so that's a plus.

**Comment by brenton** [comment]
@bfallonf  The lower level documentation I was referring to is https://github.com/openshift/origin/blob/master/docs/builds.md#how-it-works.  Personally I think that's out of scope for the core concepts and developer docs.  The developers docs in this case are for people developing on OpenShift.  This is more of a concern for someone developing OpenShift or an advanced topic for an administrator.  I think it would be reasonable to find a way to at least point advanced users directly to certain topics in the Origin repository docs.  If only to let them know the documentation exists and what extra information can be found there.

**Comment by bfallonf** [comment]
@brenton I can see why it'd be out of scope for the developer, but the architecture section isn't necessarily aimed for them. How the builder works is just conceptual stuff anyway, not a process or anything.

It seems a little cheating, but I've added a link to the upstream doc. If you disagree with any of this, please let me know.

**Comment by brenton** [comment]
@bfallonf, that looks good to me.

**Comment by bfallonf** [comment]
@brenton Cool. Thanks for that.

@adellape Any more thoughts?

**Comment by bfallonf** [comment]
@adellape bump

File: architecture/core_concepts/builds_and_image_streams.adoc [hunk 4]
```diff
@@ -11,11 +11,22 @@
...
-A link:../../dev_guide/builds.html#defining-a-buildconfig[BuildConfig] object is
+A link:../../dev_guide/builds.html[build] is the process of transforming input
+parameters into a resulting object. Most often, the process is used to transform
+input parameters or source code into a runnable image. A
+link:../../dev_guide/builds.html#defining-a-buildconfig[BuildConfig] object is
 the definition of the entire build process.
 
+OpenShift leverages Kubernetes by creating Docker containers from build images
+and pushing them to a
+link:../../install_config/install/docker_registry.html[Docker registry].
```
Comment (adellape) [inline architecture/core_concepts/builds_and_image_streams.adoc]: The Architecture topic on the integrated registry (https://docs.openshift.com/enterprise/latest/architecture/infrastructure_components/image_registry.html#integrated-openshift-registry) may be a better target here.

**Comment by adellape** [comment]
@bfallonf One comment inline, otherwise LGTM after a hearty squash.

**Comment by bfallonf** [comment]
@adellape Done. Ready to merge if you agree.

**Comment by adellape** [comment]
:+1:

**Pull request merged**